pkg/sql/alter_database.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package sql
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/sqlbase"
+)
+
+type alterDatabaseOwnerNode struct {
+	n    *tree.AlterDatabaseOwner
+	desc *sqlbase.MutableDatabaseDescriptor
+}
+
+// AlterDatabaseOwner transforms a tree.AlterDatabaseOwner into a plan node.
+func (p *planner) AlterDatabaseOwner(
+	ctx context.Context, n *tree.AlterDatabaseOwner,
+) (planNode, error) {
+	// TODO(lucy-zhang): Switch this over to the mutable db access method.
+	immutableDbDesc, err := p.ResolveUncachedDatabaseByName(ctx, n.Name.String(), true /*required*/)
+	if err != nil {
+		return nil, err
+	}
+
+	dbDesc := sqlbase.NewMutableExistingDatabaseDescriptor(*immutableDbDesc.DatabaseDesc())
+	privs := dbDesc.GetPrivileges()
+
+	hasOwnership, err := p.HasOwnership(ctx, dbDesc)
```

Comment:
you don't need this check now right? Or why can't the `canAltertoNewOwner` check this

Reply:
I made `canAltertoNewOwner` take a bool instead of explictly checking the descriptor for has ownership. Since for the alter table case, we check before executing the alter table owner (the check happens when creating the alterTable node) and reuse the bool after.

Reply:
ok